docs/Installation/QuickStartWithDockerAndVSCode.md

```diff
@@ -0,0 +1,567 @@
+the repository to your GitHub account, then clone your fork. From the
+command line, you can enter the command
+
+~~~~
+git clone git@github.com/YOUR_GITHUB_USER_NAME/spectre.git
+~~~~
+
+replacing `YOUR_GITHUB_USER_NAME` with your GitHub user name.
+
+If you're new to git, GitHub has [extensive
+documentation](https://help.github.com/en/github/using-git), including help with
+forking a repo at
+(https://help.github.com/en/github/getting-started-with-github/fork-a-repo).
+
+## Create a SpECTRE docker container
+
+From the command line (`/Applications/Terminal.app` on macOS, the Command Prompt
+or Power Shell on Windows, or the terminal/shell in Linux), create a Docker
+container for SpECTRE.
+
+A container acts kind of like a virtual machine running a known Linux
+environment. By using the SpECTRE docker container, you're free from worrying
+about installing SpECTRE's prerequisites on your platform.
+
+Here's an example command to create the container. Before running it, you'll
+want to edit it.
+
+~~~~
+docker run -p 4444:4444 \
+-v /Users/geoffrey/Codes/spectre/spectre-alt:\
+/Users/geoffrey/Codes/spectre/spectre-alt:delegated \
+-v /Users/geoffrey/Dropbox\ \(Personal\)/:/work/dropbox:delegated \
+-v /Users/geoffrey/Documents:/work/documents:delegated \
+--name spectre_vscode -i -t sxscollaboration/spectrebuildenv:latest \
+/bin/bash -l
+~~~~
+
+You'll want to edit parts of this command:
+  * `-p 4444:4444` forwards port 4444, in case you'd like to run jupyter
+    notebooks from inside your docker container. If you don't plan to do this,
+    you can omit this option.
+  * In the option `-v` option containing
+    `/Users/geoffrey/Codes/spectre/spectre-alt`,
+    replace each occurence of
+    `/Users/geoffrey/Codes/spectre/spectre-alt` with the path where you
+    cloned your SpECTRE fork. ***If you are not using macOS, omit `:delegated`
+    at the end of this option. This option improves file system performance.***
+  * This command includes a few other `-v` options. Feel free to add a `-v`
+    option to link any paths you might wish to have accessible from inside the
+    Docker container. If you don't wish to make paths (besides your SpECTRE
+    clone's path) accessible inside the container, you can omit these extra `-v`
+    options. ***Omit `:delegate` unless you are running macOS.***
+  * Feel free to change the name of the container, if you wish, by changing
+    `spectre_vscode` in the option `--name spectre_vscode`.
+
+After running this command, you'll get a command prompt inside the container,
+running as root. Enter the following commands:
+
+~~~~
+# Create the directory where you will build spectre
+mkdir /work/spectre-build-clang
+
+# Install extra python packages in your container
+apt-get install python3-pip
+pip3 install autopep8 pylint jupyter notebook scipy numpy matplotlib
+pip install autopep8 pylint jupyter notebook scipy numpy matplotlib
+
+exit
+~~~~
+
+## Configure a Visual Studio Code workspace for your container
```

Comment:
- I have been working with a `.devcontainer.json` file, have you tried that? It's explained in detail here: https://code.visualstudio.com/docs/remote/containers
- I think we should add a `.devcontainer.json` to the repository and a `.vscode` config to our docker container. Then, when a new user opens the spectre repository in vscode and has the "Remote Containers" extension installed, they will be presented with the option to re-load the folder in our Docker container. It will be automatically pulled and started with the options we have configured in the `.devcontainer.json`, so the user will be dropped in the fully configured environment right away. It even forwards ssh-agents and git configuration from the host. I comment below on a few details.
- I expect this can be very helpful for new students or when we run a workshop. What do you think?